packages/sitecore-jss/src/graphql-request-client.ts

```diff
@@ -13,22 +14,20 @@ export class GraphQLRequestClient {
    * @param {Object} variables graphql variables
    */
   async request<T>(query: string, variables?: { [key: string]: unknown }): Promise<T> {
-    const result = await this.requestInternal<T>(query, variables);
-
-    return result;
-  }
-
-  /**
-   * Create new graphql request client and execute request
-   * @param {string} query graphql query
-   * @param {Object} variables graphql variables
-   */
-  private requestInternal<T = unknown>(
-    query: string,
-    variables?: { [key: string]: unknown }
-  ): Promise<T> {
     const client = new GraphQLClient(this.endpoint);
+    const onError = (error: unknown) => {
+      console.error(
+        chalk.red(`
+          Error occurred while fetching attempting to fetch graphQL data.
+          Endpoint: ${this.endpoint}
+          Query: ${query}
+          Variables: ${JSON.stringify(variables, null, 2)}
+          Error: ${JSON.stringify(error, null, 2)}
+        `)
+      );
+    };
 
-    return client.request(query, variables);
+    // todo: we should also print the query if we are in debug mode (Anastasiya, March 2021)
+    return await client.request(query, variables).catch(onError);
```

Comment:
This is not a new addition. We were using `await` before. Removing `await` would be a breaking change as it would change the return type

Reply:
@anastasiya29 But previously we had
`return client.request(query, variables);` 
and now we have
`return await client.request(query, variables).catch(onError);`
So, basically in both cases you are returning promise. I believe that if you remove `await` it will not change `return type`

Reply:
This is the line where `await` was used in the pre-refactored code.
https://github.com/Sitecore/jss/blob/1fa6304b39cb2b509726be0a155c7cde8647bcf9/packages/sitecore-jss/src/graphql-request-client.ts#L16

Now, upon looking at this more closely, the await was probably not needed in the first place. I changing how the graphql client works is outside the scope of this task and would increase the QA needed, so I will add it to the refactoring list.